Building code dependent on named imports fails in 3.16.0

### Environment

N/A

### Reproduction

https://stackblitz.com/edit/github-pqgjqen4?file=demo.mjs

### Describe the bug

Code that utilizes [named imports/exports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#named_import) that have the string syntax to declare arbitrary (non js identifier) names, fail to build with nuxt 3.16.0 (dev mode works). Build works when downgrading back to 3.15.4.

```js
const a = 10

export { a as 'name.with.dots' }
```

```vue
<template>
  <div>
    <NuxtRouteAnnouncer />
    <NuxtWelcome />
  </div>
</template>

<script lang="js">
import * as demo from './demo.mjs'

console.log(demo['name.with.dots'])
</script>

```

### Additional context

_No response_

### Logs

```shell-script
> build
> nuxt build

Nuxt 3.16.0 with Nitro 2.11.5                                                                                                                                                                                                                                           nuxi  18:04:46
ℹ Building for Nitro preset: node-server                                                                                                                                                                                                                               nuxi  18:04:46
ℹ Building client...                                                                                                                                                                                                                                                         18:04:47
ℹ vite v6.2.1 building for production...                                                                                                                                                                                                                                     18:04:47
ℹ ✓ 128 modules transformed.                                                                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/manifest.json                   1.06 kB │ gzip:  0.29 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/error-500.dGVH929u.css    1.88 kB │ gzip:  0.72 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/error-404.C3V-3Mc4.css    3.56 kB │ gzip:  1.10 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/entry.ofCu9KLV.css       12.41 kB │ gzip:  2.57 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/z7DXRpCd.js               3.39 kB │ gzip:  1.53 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/DUuvXmrT.js               9.42 kB │ gzip:  3.76 kB                                                                                                                                                                                                 18:04:48
ℹ .nuxt/dist/client/_nuxt/y94aZ4V3.js             220.05 kB │ gzip: 63.21 kB                                                                                                                                                                                                 18:04:48
ℹ ✓ built in 906ms                                                                                                                                                                                                                                                           18:04:48
✔ Client built in 921ms                                                                                                                                                                                                                                                      18:04:48
ℹ Building server...                                                                                                                                                                                                                                                         18:04:48
ℹ vite v6.2.1 building SSR bundle for production...                                                                                                                                                                                                                          18:04:48
ℹ ✓ 53 modules transformed.                                                                                                                                                                                                                                                  18:04:48

 ERROR  ✗ Build failed in 350ms                                                                                                                                                                                                                                               18:04:48


 ERROR  Nuxt Build Error: [vite:esbuild-transpile] Transform failed with 1 error:                                                                                                                                                                                       nuxi  18:04:48
demo.mjs:3:14: ERROR: Using the string "name.with.dots" as an export name is not supported in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
file: demo.mjs:3:14

Using the string "name.with.dots" as an export name is not supported in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
1  |  const a = 10;
2  |
3  |  export { a as "name.with.dots" };
   |                ^


    demo.mjs:3:14: ERROR: Using the string "name.with.dots" as an export name is not supported in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
    file: demo.mjs:3:14

    Using the string "name.with.dots" as an export name is not supported in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
    1  |  const a = 10;
    2  |
    3  |  export { a as "name.with.dots" };
    |                ^

    at failureErrorWithLog (node_modules\esbuild\lib\main.js:1477:15)
    at node_modules\esbuild\lib\main.js:756:50
    at responseCallbacks.<computed> (node_modules\esbuild\lib\main.js:623:9)
    at handleIncomingPacket (node_modules\esbuild\lib\main.js:678:12)
    at Socket.readFromStdout (node_modules\esbuild\lib\main.js:601:7)
    at Socket.emit (node:events:524:28)
    at addChunk (node:internal/streams/readable:561:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
    at Readable.push (node:internal/streams/readable:392:5)
    at Pipe.onStreamRead (node:internal/stream_base_commons:189:23)

```

While bisecting nightlies, I found that the addition of `preserveModules: true` in https://github.com/nuxt/nuxt/pull/31235 seems to have caused this.

After thinking about it a little, I suppose setting something like `node18` as default target for server could work as that is the minimum supported version.

---

Interestingly `preserveModules: true` seems to also have broken Sentry module when `node --import` is used to [load the server side part](https://docs.sentry.io/platforms/javascript/guides/nuxt/#server-side). But that is an entirely different issue and the workaround from above does not apply. Just pointing it out in case this is useful for anyone looking.

<details>

```
node --import ./.output/server/sentry.server.config.mjs  .\.output\server\index.mjs
node:internal/modules/esm/resolve:275
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '<redacted>\.output\server\node_modules\_sentry\nuxt\build\esm\server\sdk.mjs' imported from <redacted>\.output\server\sentry.server.config.mjs
    at finalizeResolution (node:internal/modules/esm/resolve:275:11)
    at moduleResolve (node:internal/modules/esm/resolve:860:10)
    at defaultResolve (node:internal/modules/esm/resolve:984:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:685:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:634:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:617:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:273:38)
    at ModuleJob._link (node:internal/modules/esm/module_job:135:49) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///<redacted>/.output/server/node_modules/_sentry/nuxt/build/esm/server/sdk.mjs'
}

Node.js v22.14.0
```

</details>

--- a/packages/vite/src/server.ts
+++ b/packages/vite/src/server.ts
@@ -90,7 +90,6 @@ export async function buildServer (ctx: ViteBuildContext) {
           new RegExp('^' + escapeStringRegexp(withTrailingSlash(resolve(ctx.nuxt.options.rootDir, ctx.nuxt.options.dir.shared)))),
         ],
         output: {
-          preserveModules: true,
           entryFileNames: '[name].mjs',
           format: 'module',
           generatedCode: {

--- a/test/bundle.test.ts
+++ b/test/bundle.test.ts
@@ -58,7 +58,7 @@ describe.skipIf(process.env.SKIP_BUNDLE_SIZE === 'true' || process.env.ECOSYSTEM
     const serverDir = join(rootDir, '.output/server')
 
     const serverStats = await analyzeSizes(['**/*.mjs', '!node_modules'], serverDir)
-    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"189k"`)
+    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"190k"`)
 
     const modules = await analyzeSizes(['node_modules/**/*'], serverDir)
     expect.soft(roundToKilobytes(modules.totalBytes)).toMatchInlineSnapshot(`"1385k"`)
@@ -95,7 +95,7 @@ describe.skipIf(process.env.SKIP_BUNDLE_SIZE === 'true' || process.env.ECOSYSTEM
     const serverDir = join(rootDir, '.output-inline/server')
 
     const serverStats = await analyzeSizes(['**/*.mjs', '!node_modules'], serverDir)
-    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"540k"`)
+    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"541k"`)
 
     const modules = await analyzeSizes(['node_modules/**/*'], serverDir)
     expect.soft(roundToKilobytes(modules.totalBytes)).toMatchInlineSnapshot(`"78.4k"`)
@@ -117,7 +117,7 @@ describe.skipIf(process.env.SKIP_BUNDLE_SIZE === 'true' || process.env.ECOSYSTEM
     const serverDir = join(pagesRootDir, '.output/server')
 
     const serverStats = await analyzeSizes(['**/*.mjs', '!node_modules'], serverDir)
-    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"280k"`)
+    expect.soft(roundToKilobytes(serverStats.totalBytes)).toMatchInlineSnapshot(`"281k"`)
 
     const modules = await analyzeSizes(['node_modules/**/*'], serverDir)
     expect.soft(roundToKilobytes(modules.totalBytes)).toMatchInlineSnapshot(`"1396k"`)

